File: clients/py/hello_client.py
#!/usr/bin/env python

import sys
sys.path.append('gen-py')

import hello
from hello import HelloService
from hello.ttypes import HelloMsg

from thrift.transport import TTransport
from thrift.transport import TSocket
from thrift.protocol import TBinaryProtocol

host = 'localhost'
port = 8001
text = 'from Python'
try:
    text = sys.argv[1]
except:
    pass

transport = TTransport.TFramedTransport(TSocket.TSocket(host, port))
client = HelloService.Client(TBinaryProtocol.TBinaryProtocol(transport))
transport.open()
try:
    msg = client.sayHello(HelloMsg(text))
    print(msg.name)
finally:
    transport.close()

File: clients/rb/hello_client.rb
#!/usr/bin/env ruby

$:.push('gen-rb')

require 'rubygems'
require 'hello_service'

begin
  socket = Thrift::Socket.new('localhost', 8001)
  transport = Thrift::FramedTransport.new(socket)
  protocol = Thrift::BinaryProtocol.new(transport)
  client = HelloService::Client.new(protocol)
  transport.open()

  client.ping()

  msg = HelloMsg.new()
  msg.name = "from Ruby"
  msg = client.sayHello(msg)
  puts msg.name

  transport.close()

rescue Thrift::Exception => tx
  print 'Thrift::Exception: ', tx.message, "\n"
end
